feat(app): App robot settings temp netwroking

koji opened the pull request:
<!--
Thanks for taking the time to open a pull request! Please make sure you've read the "Opening Pull Requests" section of our Contributing Guide:

https://github.com/Opentrons/opentrons/blob/edge/CONTRIBUTING.md#opening-pull-requests

To ensure your code is reviewed quickly and thoroughly, please fill out the sections below to the best of your ability!
-->

# Overview
This will fix #8762 partially because the current design is missing a couple of functions' details. Need to talk about them with the design team.
I will add notes for missing pieces on Figma. 
Network connectivity information   Wifi(ip address, subnet mask, mac address) &  USB(ip address, subnet mask, mac address) are implemented in this PR.

USB-to-Ethernet information has been moved into AppSettings Advanced already. 

The design has been updated.
https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/Milestone-1%3A-Read-and-Execute?node-id=1297%3A105705#178196044


<!--
Use this section to describe your pull-request at a high level. If the PR addresses any open issues, please tag the issues here.
-->

# Changelog
- Added Wireless network information
- Added Wired network information
- Added Disconnect from Wi-Fi network button

<!--
List out the changes to the code in this PR. Please try your best to categorize your changes and describe what has changed and why.

Example changelog:
- Fixed app crash when trying to calibrate an illegal pipette
- Added state to API to track pipette usage
- Updated API docs to mention only two pipettes are supported

IMPORTANT: MAKE SURE ANY BREAKING CHANGES ARE PROPERLY COMMUNICATED
-->

# Review requests
- Display Wi-Fi info and Wired USB info (RobotSettingsNetworking)
- Most codes are taken from 5.0.2 except RobtSettingsNetworking so they should be okay.

<!--
Describe any requests for your reviewers here.
-->

# Risk assessment
low
<!--
Carefully go over your pull request and look at the other parts of the codebase it may affect. Look for the possibility, even if you think it's small, that your change may affect some other part of the system - for instance, changing return tip behavior in protocol may also change the behavior of labware calibration.

Identify the other parts of the system your codebase may affect, so that in addition to your own review and testing, other people who may not have the system internalized as much as you can focus their attention and testing there.
-->

koji committed:
Initial styling and disconnect function

koji committed:
Added slideoutbase

koji committed:
Modified Networking tab styling & added NetworkingSlideout

koji committed:
update styling

koji committed:
Clean up Networking Tab

koji committed:
Copied existing SelectNetwork code for Networking slideout

koji committed:
sync with edge branch
Merge branch 'edge' into app_robot_setting_networking

koji committed:
Added existing SelectNetwork component to Networking tab

This should be a temporary patching for Networking functionality. Once the team get the details of Networking functionality, the team will start implement this functionality.

koji committed:
Moved existing test cases

koji committed:
fixed import statement order

koji committed:
Fixed test error

koji committed:
Added a test for Networking tab on RobotSettings

koji committed:
Added test for RobotSettingsNetworking and modified codes for running test cases

koji committed:
Added test cases for displaying wifi and usb info

koji committed:
fixed lint-js issues

koji committed:
Added Wi-Fi network name display cases

koji committed:
fixed import path

koji committed:
Merge branch 'app_RobotSettings_temp_netwroking' of https://github.com/Opentrons/opentrons into app_RobotSettings_temp_netwroking

koji committed:
fix stupid typo

koji committed:
remove Disconnect from Wi-Fi button

koji committed:
Added the label to align with the updated design

koji committed:
Updated test cases to align with updated design

b-cooper commented on app/src/organisms/Devices/RobotSettings/__tests__/RobotSettingsNetworking.test.tsx:
```diff
@@ -0,0 +1,202 @@
+
+const mockWifiList = [
+  { ...Fixtures.mockWifiNetwork, ssid: 'foo', active: true },
+  // { ...Fixtures.mockWifiNetwork, ssid: 'bar' },
```
why is this fixture commented out?

koji replied:
I forgot to remove it.  RobotNetworkingTab finds active net only otherwise the network name is null.

koji replied:
Updated the test case.

b-cooper reviewed (approved):
The networking tab looks good 👍 .  The copypasta select dropdown will be an effective temporary solution.

koji committed:
Updated the test case

prev version:  passed one wifi network object (active wifi)
updated version: passed two wifi network object (active wifi and non-active wifi)
Here active network is the network that a robot is connecting.
